### PR TITLE
[slikenet] fix wasm32-emscripten build

### PR DESCRIPTION
**Pull request by Pospelove**
Upstreaming the fix I did for my project. It works ([logs](https://github.com/skyrim-multiplayer/skymp/actions/runs/10603980733/job/29389544123))

Just `#ifdef Emscripten` everywhere we `#ifdef Linux` and we're good

### CODE REVIEW

**Comment by jimwang118** [comment]
All features passed with following triplets:

```
x64-windows
x64-linux
```

File: ports/slikenet/fix-emscripten.patch [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
++++ b/Source/include/slikenet/Export.h
+@@ -15,7 +15,7 @@
+ 
+ #include "defines.h"
+ 
+-#if defined(_WIN32) && !(defined(__GNUC__)  || defined(__GCCXML__)) && !defined(_RAKNET_LIB) && defined(_RAKNET_DLL)
++#if defined(_WIN32) && !(defined(__EMSCRIPTEN__) || defined(__GNUC__)  || defined(__GCCXML__)) && !defined(_RAKNET_LIB) && defined(_RAKNET_DLL)
+ #define RAK_DLL_EXPORT __declspec(dllexport)
+ #else
+ #define RAK_DLL_EXPORT
```
Comment (dg0yt) [inline ports/slikenet/fix-emscripten.patch:13]:
```suggestion
```
`defined _WIN32` already excludes emscripten.

File: ports/slikenet/fix-emscripten.patch [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
++++ b/Source/src/PacketLogger.cpp
+@@ -504,7 +504,7 @@ void PacketLogger::SetSuffix(const char *_suffix)
+ }
+ void PacketLogger::GetLocalTime(char buffer[128])
+ {
+-#if defined(_WIN32) && !defined(__GNUC__)  && !defined(__GCCXML__)
++#if defined(_WIN32) && !defined(__GNUC__)  && !defined(__GCCXML__) && !defined(__EMSCRIPTEN__)
+     time_t rawtime;
+ 	struct timeval tv;
+ 	// If you get an arror about an incomplete type, just delete this file
```
Comment (dg0yt) [inline ports/slikenet/fix-emscripten.patch:130]:
```suggestion
```
`defined _WIN32` already excludes emscripten.

File: ports/slikenet/fix-emscripten.patch [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
++++ b/Source/src/gettimeofday.cpp
+@@ -13,7 +13,7 @@
+  *  license found in the license.txt file in the root directory of this source tree.
+  */
+ 
+-#if defined(_WIN32) && !defined(__GNUC__)  &&!defined(__GCCXML__)
++#if defined(_WIN32) && !defined(__GNUC__)  &&!defined(__GCCXML__) && !defined(__EMSCRIPTEN__)
+ 
+ #include "slikenet/gettimeofday.h"
+
```
Comment (dg0yt) [inline ports/slikenet/fix-emscripten.patch:234]:
```suggestion
```
`defined _WIN32` already excludes emscripten.

**Review by dg0yt — CHANGES_REQUESTED** [review]
The changes shall be upstreamed so that the patch can be removed in future versions of the port.

**Pull request merged**